### DISCUSSION TITLE
Subscription & Notification Examples For All Three Subscription types

### DISCUSSION THREAD

**Discussion by sonivi** [ecederstrand/exchangelib#1098] [Q&A]
Hi Team,

Please help me with Python Exchagelib samples example on how to create a subscription in case of any EventType like NewMail, etc for all three three subscription types for a mailbox.

I tried with code snippet mentioned in https://ecederstrand.github.io/exchangelib/#synchronization-subscriptions-and-notifications for subscribe to pull notification but did not get any notification via code when I manually sent a new email to that mailbox.

Also tried subscribe to push notification but when tried to get notification via code when I manually sent an new email to that mailbox. I got an error stating subscription ID is invalid. I got subscription ID with account.inbox.subscribe_to_push() function

**Comment by ecederstrand** [comment]
There are some more examples in the test suite showing working subscription code. For example: https://github.com/ecederstrand/exchangelib/blob/6c4a0fe1667b8a115fc26e8c46151c7473efc9a3/tests/test_items/test_sync.py#L222

**Reply by sonivi** [reply]
I tried below code but getting notifications count as one, before and after getting events.
when code is sleeping for 120 seconds. I am creating a new email in drafts folder manually so ideally as per my understanding count of notifications should be increased to whatever count I get before sleep.

```python
from exchangelib import Credentials, Configuration
from exchangelib import Credentials, Configuration
from exchangelib import DELEGATE, Account
import time

credentials = Credentials(username=r'domain\user', password='password')
config = Configuration(credentials=credentials, service_endpoint='https://abc.com/EWS/Exchange.asmx')
a = Account(
  primary_smtp_address='user@abc.com', config=config,
  autodiscover=False, access_type=DELEGATE,
)

test_folder = a.drafts
with test_folder.pull_subscription() as (subscription_id, watermark):
    notifications = list(test_folder.get_events(subscription_id, watermark))
    print(f"Before.. notification count {len(notifications)}")
    
    print("Sleeping for 120 seconds....")
    time.sleep(120)
    
    notifications = list(test_folder.get_events(subscription_id, watermark))
    print(f"After.. notification count {len(notifications)}")
```

**Reply by ecederstrand** [reply]
That's supposed to work, at a glance. I don't know why it doesn't work for you on your server.
